src/catalog/src/memory/objects.rs

```diff
@@ -652,40 +652,36 @@ pub struct Log {
 
 #[derive(Debug, Clone, Serialize)]
 pub struct Sink {
-    pub create_sql: String,
-    pub from: GlobalId,
-    pub connection: StorageSinkConnection<ReferencedConnection>,
-    // TODO(guswynn): this probably should just be in the `connection`.
-    pub envelope: SinkEnvelope,
+    pub sink: mz_sql::plan::Sink,
```

Comment:
The justification is indeed not in this PR. I included this change because I had already typed it and was standalone.

The reason I had to do this was because I started implementing `fn plan_alter_sink` which needs to get the current sink from the catalog and check whatever needs to be checked. So I started by adding a `fn sink(&self) -> Option<&mz_sql::plan::Sink>` in the `CatalogItem` trait of the `mz_sql` crate but when I tried to implement it in the `mz_catalog` crate I faced a problem because there we store a `mz_catalog::memory::objects::Sink` which is not available to `mz_sql` since that leads to a circular dependency.

Here are the options I considered:
1. Do what I have here since the `mz_catalog` type is a superset of the `mz_sql` one. That makes it easy to return a reference to the right type. It looks like we do the same thing with function items where we return an `&mz_sql::func::Func` from `mz_catalog`.
2. Return an `Option<mz_sql::plan::Sink>` instead (no reference) and construct an instance of this on the fly. This is maybe a bit more flexible but it still requires the catalog to be able to construct an `mz_sql::plan::Sink` instance on the fly so it must have all the information at hand.
3. Return an `Option<mz_storage_types::StorageSinkDesc>`. This is similar to how `CatalogItem::source_desc` works, effectively going through a third crate to solve the circular dependency. It feels weird to me to rely on storage specific types at this layer so I didn't go down this path.

Is there a better way to go about this?

Reply:
For posterity: we discussed on Slack [here](https://materializeinc.slack.com/archives/C02FWJ94HME/p1716821078686919?thread_ts=1716820581.458629&cid=C02FWJ94HME), and @petrosagg now thinks this won't be necessary.

Reply:
This is now implemented as discussed in slack